src/amici.cpp

```diff
@@ -1383,6 +1384,375 @@ int getDiagnosisB(int it, void *ami_mem, UserData *udata, ReturnData *rdata, Tem
     return status;
 }
 
+/* ------------------------------------------------------------------------------------- */
+/* ------------------------------------------------------------------------------------- */
+/* ------------------------------------------------------------------------------------- */
+
+
+int applyNewtonsMethod(void *ami_mem, UserData *udata, ReturnData *rdata, TempData *tdata, int status, int ntry) {
+    /**
+     * applyNewtonsMethod applies Newtons method to the current state x to find the steady state
+     *
+     * @param[in] ami_mem pointer to the solver memory block @type *void
+     * @param[in] udata pointer to the user data struct @type UserData
+     * @param[in] tdata pointer to the temporary data struct @type TempData
+     * @param[out] tdata pointer to the temporary data struct @type TempData
+     * @param[out] rdata pointer to the return data struct @type ReturnData
+     * @return void
+     */
+
+    int ix;
+    int inewt = 0;
+    int stat1 = -1;
```

Comment:
What is `stat1`? If status, please name so and use descriptive status code from `amici_defines.h` or add a new one if appropriate.